packages/std/src/storage.rs

```diff
@@ -18,8 +19,8 @@ impl MemoryStorage {
 }
 
 impl ReadonlyStorage for MemoryStorage {
-    fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
-        self.data.get(key).cloned()
+    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>> {
```

Comment:
Since this implementation always returns `Ok` and is what we use for Mock, maybe we need another mock as well, that always returns error (so we can test this is handled properly)?

That said, we always just pass errors up (`?`) and not much to test there

Reply:
Well, maybe some storage wrappers could use this to ensure they don't ignore errors

Reply:
Rust ensure a Result's error is not ignored, at least by warnings.

If you really want to test this for the mock implementation, one simple approach would be to limit the number of elements to X. This would allow for error reporting at least in `.set`